### ISSUE TITLE
Check if files exist before downloading

### ISSUE THREAD

**Issue by faroit** [ynop/audiomate#125]
Running

```python
esc_downloader = io.ESC50Downloader()
esc_downloader.download('esc')
esc_downloader.download('esc')
```

downloads a dataset twice. Whereas the second time the download is completed not saved to the target folder because the folder already exists.

```
Download File: 647Mit [01:24, 7.65Mit/s]
Download File: 647Mit [01:35, 6.79Mit/s]
Traceback (most recent call last):
  File "test.py", line 7, in <module>
    esc_downloader.download('esc')
  File "/Users/faro/repositories/audiomate_test/env/lib/python3.6/site-packages/audiomate/corpus/io/base.py", line 28, in download
    return self._download(target_path)
  File "/Users/faro/repositories/audiomate_test/env/lib/python3.6/site-packages/audiomate/corpus/io/downloader.py", line 52, in _download
    delete_subfolders=True
  File "/Users/faro/repositories/audiomate_test/env/lib/python3.6/site-packages/audiomate/utils/files.py", line 29, in move_all_files_from_subfolders_to_top
    shutil.move(src, target)
  File "/Users/faro/repositories/audiomate_test/env/lib/python3.6/shutil.py", line 548, in move
    raise Error("Destination path '%s' already exists" % real_dst)
shutil.Error: Destination path 'esc/tests/tests' already exists
```

* either check the existence of a root path before downloading any files at least raise a warning.
* or add a `force_redownload` flag  that would forcefully redownload the dataset (even if already existing)

this issue is part of a [JOSS review](https://github.com/openjournals/joss-reviews/issues/2135)

**Comment by ynop** [comment]
You are right.
I agree on both your suggestions:
- Fail if destination exists
- Optional flag to overwrite

**Comment by cdkini** [comment]
@ynop do you know if anyone has started on this? I'd like to work on it if not

**Comment by ynop** [comment]
Not started yet, would be nice if you help out.

**Comment by cdkini** [comment]
Is the issue specific to ESC50Downloader or should the change affect all classes that inherit from ArchiveDownloader? 

I'd imagine the issue is prevelant for all such downloader classes but I want to make sure before changing the parent class.

**Comment by ynop** [comment]
This should apply for all downloaders, so you can implement directly in `CorpusDownloader`.

**Comment by ynop** [comment]
Fixed by #128
